services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/logging/CheckLoggingSpec.groovy

```diff
@@ -1,9 +1,89 @@
 package org.openkilda.functionaltests.spec.logging
 
+import groovy.util.logging.Slf4j
 import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.service.elastic.ElasticQueryBuilder
+import org.openkilda.testing.service.elastic.ElasticService
+import org.openkilda.testing.service.elastic.model.KildaTags
+import org.openkilda.testing.service.elastic.model.KildaTypes
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.web.client.HttpClientErrorException
 import spock.lang.Narrative
 
+@Slf4j
 @Narrative("This specification ensures that all logging facilities are up and running after Kilda deployment")
 class CheckLoggingSpec extends BaseSpecification {
 
+    @Autowired
+    ElasticService elasticService
+
+    String discoveryMessage = "push discovery package via"
+
+    def "Check Floodlight logging"() {
+        when: "Retrieve floodlight logs for last 5 minutes"
+        def result
+        if (this.profile.equalsIgnoreCase("virtual")) {
+            result = elasticService.getLogs(new ElasticQueryBuilder().setField("type", KildaTypes.FLOODLIGHT).
+                    setTimeRange(300).build())
+        } else {
+            result = elasticService.getLogs(new ElasticQueryBuilder().setTags(KildaTags.FLOODLIGHT).
+                    setTimeRange(300).build())
+        }
+        assert result?.hits?.total > 0
+
+        then: "There should be discovery messages"
+        result.hits.hits.any {hit -> hit.source.message.toLowerCase().contains(discoveryMessage)}
+    }
+
+    def "Check Northbound, Storm and Topology Engine logging"() {
+        setup:
```

Comment:
This setup doesn't have any sense. Please move 

int timeout = 60

under 'when' block.

Also, pay attention that we use 'given' instead of 'setup'. There is no difference between them but let's have consistency.